### ISSUE TITLE
Demo app for Vue

### ISSUE THREAD

**Issue by asharghi** [triniwiz/nativescript-stripe#83]
Thank you for making such a great plugin. The CreditCardView looks good but I'm not able to make the StandardComponent work with Vue. Not sure where and how to do the StripeService.createPaymentSession(..). Is there any possibility of getting a demo for Vue?

Edit: StackOverflow question with more details: https://stackoverflow.com/questions/57735258/how-to-setup-nativescript-stripe-in-nativescript-vue

**Comment by asharghi** [comment]
I am finally able to run the app with this setup in my Vue component: 

```
import { StripeService, Listener } from "~/shared/stripe/stripe.service.ts";
const Page = require("tns-core-modules/ui/page").Page;

let stripeService = new StripeService();
let paymentSession = null;

export default {
  mounted() {
    paymentSession = stripeService.createPaymentSession(
      new Page(),
      this.item.price,
      new Listener(this)
    );
  },
 data() {
    return {
      item: {
        id: 0,
        name: "Something to buy",
        price: 1200
      },
      paymentInProgress: false,
      canBuy: true,
      isLoading: false,
      paymentType: "",
      paymentImage: "",
      shippingType: "",
      total: "",
      debugInfo: "",
      successMessage: "",
      errorMessage: ""
    };
  },
  methods: {
    showPaymentMethods() {
      return stripeService.showPaymentMethods(paymentSession);
    },
    showShipping() {
      return stripeService.showShipping(paymentSession);
    },
    buy() {
      this.paymentInProgress = true;
      this.canBuy = false;
      return stripeService.requestPayment(paymentSession);
    }
  },...
```

In my _stripe.service.ts_ I have the same code as the Angular demo, but I have also added the Listener class there: 

```
export class Listener {
  
  public component;
  constructor(component) {
      this.component = component;
  }
  onCommunicatingStateChanged(_isCommunicating) {
      
  }
  onPaymentDataChanged(data) {
      this.component.paymentMethod = data.paymentMethod;
      this.component.shippingInfo = data.shippingInfo;
      this.component.shippingAddress = data.shippingAddress;
  }
  onPaymentSuccess() {
      this.component.successMessage =
          `Congratulations! You bought a "${this.component.item.name}" for $${this.component.item.price / 100}.`;
      
  }
  onUserCancelled() {
  }
  onError(_errorCode, message) {
      this.component.errorMessage = message;
  }
  provideShippingMethods(address) {
      let upsGround = {
          amount: 0,
          label: "UPS Ground",
          detail: "Arrives in 3-5 days",
          identifier: "ups_ground"
      };
      let upsWorldwide = {
          amount: 1099,
          label: "UPS Worldwide Express",
          detail: "Arrives in 1-3 days",
          identifier: "ups_worldwide"
      };
      let fedEx = {
          amount: 599,
          label: "FedEx",
          detail: "Arrives tomorrow",
          identifier: "fedex"
      };
      let methods = {};
      if (!address.country || address.country === "US") {
          methods['isValid'] = true;
          methods['validationError'] = undefined;
          methods['shippingMethods'] = [upsGround, fedEx];
          methods['selectedShippingMethod'] = fedEx;
      }
      else if (address.country === "AQ") {
          methods['isValid'] = false;
          methods['validationError'] = "We can't ship to this country.";
      }
      else {
          fedEx.amount = 2099;
          methods['isValid'] = true;
          methods['validationError'] = undefined;
          methods['shippingMethods'] = [upsWorldwide, fedEx];
          methods['selectedShippingMethod'] = fedEx;
      }
      return methods;
  }
}
```
The result is this:

![uLwzbcd](https://user-images.githubusercontent.com/9407896/64062288-ca784980-cc0e-11e9-8554-12eb6afad441.png)

The issue I'm having now is that nothing happens when I tap "Payment type". When I debug, I can see it going into the method presentPaymentMethods(). And this code is run without any erros but nothing happens:

```
 StripePaymentSession.prototype.presentPaymentMethods = function () {
        this.ensureHostViewController();
        this.native.presentPaymentOptionsViewController();
    };
```

Any suggestions?

**Comment by RobertGardner** [comment]
I don't know anything about Vue so I can't debug or be certain of the problem.

Two things to check:
1- Is the plugin being initialized? On iOS in the angular demo, you get a spinning cursor in the "Select Payment" value field while the plugin initializes and then it either shows an Apple Pay icon or says "Select Payment" in light gray. I don't see that here. Also, on iOS your Listener's `onPaymentDataChanged` should get called when the plugin first initializes.

2- The native code displays the payment options in a dialog. Is the host view controller being setup properly for Vue? (step into `ensureHostViewController` to see what it does.

Good luck! If you get it working, it would be superb if you sent a PR with your Vue demo app.

**Comment by asharghi** [comment]
I got the answer to where and how to call the stripeService.createPaymentSession() now. 

On the Page tag in my template I now have:
`<Page ref="cartPage" @loaded="onPageLoaded" class="page">`

And then I moved createPaymentSession() from the mounted() event to onPageLoaded() under methods:
```
    onPageLoaded(args) {
      var comp = this;
      paymentSession = stripeService.createPaymentSession(
        args.object,
        comp.stripeItem.price,
        new Listener(comp)
      );
    }
```

Hope this helps the next guy. I will make a PR with a blank demo app app in {N}-Vue when I've tested the setup a little more.
